### ISSUE TITLE
Support release.yml

### ISSUE THREAD

**Issue by ScalaWilliam** [ActionFPS/ActionFPS-Game#40]
[release.yml](https://github.com/ActionFPS/ActionFPS-Game-Media/blob/master/release.yml)

1. Download a ZIP file.
2. Extract it in user location.
3. Set all files to non-executable for security.
4. Behave with downloaded packages similarly as we do with the `PATH` variable look up.

**Comment by ScalaWilliam** [comment]
Reference implementation:
https://github.com/ActionFPS/ActionFPS-Game-Media/blob/master/download.py

Sorry no tests right now ;-)

**Comment by ScalaWilliam** [comment]
I found this also: https://github.com/google/omaha
And looked at RDF which would be nice. Perhaps RDF can be used as a source format, the yml (or anything else) be only a projection of it. With RDF we can attach author info to various maps etc.

But ended up concluding that YAML will be the simplest. We can parse it much more easily than anything else. And if you want even-easier we can think about other formats.

The downloadpy thing is parallelizable so it'll be possible to download all the required resources in parallel.

**Comment by ScalaWilliam** [comment]
Actually, even more ideal would be to just download the ZIP and read files from inside of there directly. No point in extracting. This simplifies a LOT of the logic ;-)

**Comment by ScalaWilliam** [comment]
Walking through the code, release.yml approach seems like a proper PITA to make happen because of the inflexibility of the code. I'll try to come up with something easier.

**Comment by ScalaWilliam** [comment]
Closing in favour of https://github.com/ActionFPS/ActionFPS-Game/issues/42
